ecosystem/typescript/sdk/examples/typescript/gas_fee_payer.ts

```diff
@@ -0,0 +1,228 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+  AptosAccount,
+  FaucetClient,
+  BCS,
+  AptosClient,
+  TokenClient,
+  MaybeHexString,
+  OptionalTransactionArgs,
+  TransactionBuilder,
+  TransactionBuilderRemoteABI,
+  TxnBuilderTypes,
+  getPropertyValueRaw,
+} from "aptos";
+import { NODE_URL, FAUCET_URL } from "./common";
+import { assert } from "console";
+
+const client = new AptosClient(NODE_URL);
+const tokenClient = new TokenClient(client);
+const faucetClient = new FaucetClient(NODE_URL, FAUCET_URL);
+
+async function waitForEnter() {
+  return new Promise<void>((resolve, reject) => {
+    const rl = require("readline").createInterface({
+      input: process.stdin,
+      output: process.stdout,
+    });
+
+    rl.question("Please press the Enter key to proceed ...\n", () => {
+      rl.close();
+      resolve();
+    });
+  });
+}
+
+async function ensureTxnSuccess(txnHashPromise: Promise<string>) {
+  const txnHash = await txnHashPromise;
+  const txn = await client.waitForTransactionWithResult(txnHash);
+  assert((txn as any)?.success);
+}
+
+const getBalance = async (account: AptosAccount) => {
+  const resources = await client.getAccountResources(account.address().hex());
+  const aptosCoin = "0x1::coin::CoinStore<0x1::aptos_coin::AptosCoin>";
+  let accountResource = resources.find((r) => r.type === aptosCoin);
+  return BigInt((accountResource!.data as any).coin.value);
+};
+
+async function createTokenWithFeePayer(feePayer: AptosAccount, account: AptosAccount, collectionName: string, name: string, description: string, supply: number, uri: string, max: BCS.AnyNumber, royalty_payee_address?: MaybeHexString, royalty_points_denominator?: number, royalty_points_numerator?: number, property_keys?: Array<string>, property_values?: Array<string>, property_types?: Array<string>, extraArgs?: OptionalTransactionArgs): Promise<string> {
```

Comment:
This looks great already. But I'd add some comments at the top here to explain the overall flow. May be something along the like of:

To create a transaction with a different account paying for gas, we need to:
1. Create a fee payer raw transaction (TxnBuilderTypes.FeePayerRawTransaction) and specify which account will be paying for gas.
2. Two signatures are needed - one from the account sending the transaction and one from the separate gas payer. In this example, the sender will sign first, followed by the gas payer. However, the reverse order should also work. The signatures can be generated the same way they are for normal transactions.
3. Once we have two signatures, the transaction can be sent, and gas fees will be deducted from the gas payer account instead.

Reply:
Sounds good. I just added the description to the example.